### PR TITLE
solution

### PR DESCRIPTION
**Pull request by OleksandrDzindziura**
[DEMO LINK](https://derelicttt.github.io/react_static-list-of-todos/)

### CODE REVIEW

File: src/components/TodoList/TodoList.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <th scope="col">Name</th>
+        <th scope="col">Username</th>
+        <th scope="col">Email</th>
+        <th scope="col">Title</th>
+        <th scope="col">Status</th>
+      </tr>
+    </thead>
+    <tbody>
+      {preparedTodos.map(preparedTodo => (
+        <tr key={preparedTodo.id} className={preparedTodo.completed ? 'table-success' : ''}>
```
Comment (Anna-Tkachenko) [inline src/components/TodoList/TodoList.tsx:24]: You can use classnames library here for setting 'table-success'

**Review by Anna-Tkachenko — APPROVED** [review]
Well Done🔥

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
 import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import todos from './api/todos';
 
-// import todos from './api/todos';
-// import users from './api/users';
+import users from './api/users';
+import { PreparedTodo } from './types/PreparedTodo';
+
+const preparedTodosArr: PreparedTodo[] = todos.map(todo => {
```
Comment (danheim) [inline src/App.tsx:9]: avoid `arr` word

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+  const user = users.find(person => person.id === todo.userId) || null;
+
+  return { ...todo, user };
+});
 
 const App: React.FC = () => (
   <div className="App">
-    <h1>Static list of todos</h1>
+    <TodoList preparedTodos={preparedTodosArr} />
```
Comment (danheim) [inline src/App.tsx:17]: just todos, avoid prepared word here
```suggestion
    <TodoList todos={preparedTodos} />
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import { PreparedTodo } from '../../types/PreparedTodo';
+
+export const TodoInfo: React.FC<PreparedTodo> = ({
```
Comment (danheim) [inline src/components/TodoInfo/TodoInfo.tsx:5]: Create Props interface
https://mate-academy.github.io/fe-program/js/extra/react-typescript

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import { UserInfo } from '../UserInfo/UserInfo';
+import { PreparedTodo } from '../../types/PreparedTodo';
+import { TodoInfo } from '../TodoInfo/TodoInfo';
+
+type Props = {
+  preparedTodos: PreparedTodo[]
```
Comment (danheim) [inline src/components/TodoList/TodoList.tsx:7]: avoid prepared word

File: src/components/TodoList/TodoList.tsx (hunk 1 above)
Comment (danheim) [reply src/components/TodoList/TodoList.tsx:24]: Agree with Anna, use `classnames` library

File: src/components/TodoList/TodoList.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        <th scope="col">Email</th>
+        <th scope="col">Title</th>
+        <th scope="col">Status</th>
+      </tr>
+    </thead>
+    <tbody>
+      {preparedTodos.map(preparedTodo => (
+        <tr key={preparedTodo.id} className={preparedTodo.completed ? 'table-success' : ''}>
+          <UserInfo {...preparedTodo.user} />
+          <TodoInfo {...preparedTodo} />
```
Comment (danheim) [inline src/components/TodoList/TodoList.tsx:26]:
```suggestion
          <UserInfo user={todo.user} />
          <TodoInfo todo={todo} />
```

File: src/components/UserInfo/UserInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { User } from '../../types/User';
+
+export const UserInfo: React.FC<Partial<User>> = ({
```
Comment (danheim) [inline src/components/UserInfo/UserInfo.tsx:4]: Create Props interface
https://mate-academy.github.io/fe-program/js/extra/react-typescript

File: src/types/PreparedTodo.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { UserTodo } from './UserTodo';
+import { User } from './User';
+
+export interface PreparedTodo extends UserTodo {
```
Comment (danheim) [inline src/types/PreparedTodo.ts:4]:
```suggestion
export interface TodoWithUser extends Todo {
```

File: src/types/UserTodo.ts [hunk 9]
```diff
@@ -0,0 +1,6 @@
+export interface UserTodo {
```
Comment (danheim) [inline src/types/UserTodo.ts:1]:
```suggestion
export interface Todo {
```

**Review by danheim — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 10]
```diff
@@ -1,12 +1,20 @@
...
+const preparedTodos: Todo[] = todos.map(todo => {
+  const user = users.find(person => person.id === todo.userId) || null;
+
+  return { ...todo, user };
+});
 
 const App: React.FC = () => (
   <div className="App">
-    <h1>Static list of todos</h1>
+    <TodoList Todos={preparedTodos} />
```
Comment (danheim) [inline src/App.tsx:17]: почему с большой буквы? переменные в JS мы называем camelCase

File: src/components/TodoInfo/TodoInfo.tsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo
+}
```
Comment (danheim) [inline src/components/TodoInfo/TodoInfo.tsx:7]: не забывай точку с запятой
```suggestion
interface Props {
  todo: Todo;
}
```

File: src/components/TodoList/TodoList.tsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import classNames from 'classnames';
+import { UserInfo } from '../UserInfo/UserInfo';
+import { Todo } from '../../types/Todo';
+import { TodoInfo } from '../TodoInfo/TodoInfo';
+
+interface Props {
+  Todos: Todo[]
+}
+
+export const TodoList : React.FC<Props> = ({ Todos }) => (
```
Comment (danheim) [inline src/components/TodoList/TodoList.tsx:11]:
```suggestion
export const TodoList: React.FC<Props> = ({ todos }) => (
```

File: src/components/TodoList/TodoList.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    <tbody>
+      {Todos.map(todo => (
+        <tr
+          key={todo.id}
+          className={classNames(
+            {
+              'table-success': todo.completed,
+            },
+          )}
+        >
```
Comment (danheim) [inline src/components/TodoList/TodoList.tsx:32]:
```suggestion
        <tr
          key={todo.id}
          className={classNames({
            'table-success': todo.completed,
          })}
        >
```

File: src/components/TodoList/TodoList.tsx [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+          { todo.user ? <UserInfo user={todo.user} /> : null}
```
Comment (danheim) [inline src/components/TodoList/TodoList.tsx:33]:
```suggestion
          {todo.user ? <UserInfo user={todo.user} /> : null}
```

File: src/components/UserInfo/UserInfo.tsx [hunk 15]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { User } from '../../types/User';
+
+interface Props {
+  user: User
+}
```
Comment (danheim) [inline src/components/UserInfo/UserInfo.tsx:6]:
```suggestion
interface Props {
  user: User;
}
```

File: src/types/User.ts [hunk 16]
```diff
@@ -0,0 +1,6 @@
+export interface User {
+  id:number
```
Comment (danheim) [inline src/types/User.ts:2]:
```suggestion
  id: number
```

**Review by danheim — CHANGES_REQUESTED** [review]

**Review by danheim — APPROVED** [review]

**Pull request closed without merging**